ghcide/src/Development/IDE/Types/HscEnvEq.hs

```diff
@@ -82,6 +90,14 @@ newHscEnvEqWithImportPaths envImportPaths hscEnv deps = do
                     Maybes.Succeeded mi -> Just mi
         modIfaces <- mapMaybeM doOne targets
         return $ createExportsMap modIfaces
+
+    envVisibleModuleNames <-
+      fromRight []
+        <$> catchSrcErrors
+          dflags
+          "listVisibleModuleNames"
+          (evaluate . force $ listVisibleModuleNames dflags)
```

Comment:
We better make this lazy, since there are may be situations where we create and immediately throw away `HscEnvEq`. Benchmarks should confirm or deny.

To make it lazy, you can use `onceAsync` as done for `envPackageExports`. You could also simply make it pure by dropping `catchSrcErrors`, and move the error handling to the call site.

Reply:
I prefer handling the error right here.